tests/v3_validation/cattlevalidationtest/core/test_hosttemplate.py

```diff
@@ -0,0 +1,550 @@
+        assert ht1 is None
+    finally:
+        if ht1 is not None:
+            cleanupht(ht1, client)
+
+@if_machine_digocean
+def test_update(client):
+    create_args = {
+        "name": "test-do-ht1",
+        "driver": "digitalocean",
+        "secretValues": {
+            "digitaloceanConfig": {
+                "accessToken": access_token
+            }
+        },
+        "publicValues": {
+            "digitaloceanConfig": {
+                "image": "ubuntu-17-04-x64",
+                "region": "sfo2",
+                "size": "1gb"
+            }
+        }
+    }
+    ht0 = create_ht(create_args, client)
+    ht0.publicValues.digitaloceanConfig.size = "2gb"
+    new_name = random_str()
+    new_description = random_str()
+    ht0.name = new_name
+    ht0.description = new_description
+    ht1 = client.update(ht0, ht0)
+
+    assert ht1.name == new_name
+    assert ht1.description == new_description
+    assert ht1.publicValues.digitaloceanConfig.size == "1gb"
+    cleanupht(ht1, client)
+
+
+def hosts_in_do(name):
+    return [d for d in get_droplets() if d["name"] == name]
+
+
+def check_digitalocean_cleaned_up(host_name):
+    for i in range(5):
+        hosts = hosts_in_do(host_name)
+        if len(hosts) == 0:
+            break
+        time.sleep(2 ** i)
+    try:
+        assert 0 == len(hosts)
+    finally:
+        headers = {'Authorization': "Bearer " + access_token}
+        for droplet in hosts:
+            url = 'https://api.digitalocean.com/v2/droplets/' + \
+                  str(droplet["id"])
+            try:
+                r = requests.delete(url, headers=headers)
+            finally:
+                r.close()
+
+
+def create_host(host_create_args, client):
+    host0 = client.wait_success(client.create_host(**host_create_args),
+                                timeout=600)
+    assert host_create_args["hostname"] == host0.hostname
+    assert host0.id is not None and host0.id != ""
+    return host0
+
+
+def cleanuphost(host_to_remove, client):
+    deactivated_host = client.wait_success(host_to_remove.deactivate())
+    removed_host = client.wait_success(client.delete(deactivated_host))
+    assert 'removed' == removed_host.state
+    check_digitalocean_cleaned_up(removed_host.hostname)
+
+@if_machine_amazon
+@if_machine_digocean
+def test_create_host_wrong_provider(client):
+    create_args = {
+        "name": "test-aws-ht1",
+        "driver": "amazonec2",
+        "secretValues": {
+            "amazonec2Config": {
+                "accessKey": access_key,
+                "secretKey": secret_key
+            }
+        }
+    }
+    ht0 = create_ht(create_args, client)
+    host_name = random_str()
+    host_create_args = {
+        "hostname": host_name,
+        "hostTemplateId": ht0.id,
+        "digitaloceanConfig": {
+            "image": "ubuntu-16-04-x64",
+            "region": "sfo2",
+            "size": "1gb"
+        }
+    }
+    try:
+        create_host(host_create_args, client)
```

Comment:
If this did not throw an exception, you would not catch it. Use the pytest.raises paradigm